### ISSUE TITLE
Training killed

### ISSUE THREAD

**Issue by slsanna** [AlexeyAB/darknet#3781]
Hi everyone,
I'm trying to use your guide to train my model using yolo tiny. I followed all the guide, put the file in the right directory but when I type this command: ./darknet detector train ../data/obj.data ../cfg/yolov3-tiny_obj.cfg ../cfg/yolov3-tiny.conv.15
after the first resizing the process kills. I have read some issues like mine but I wasn't able to solve maybe because I tried to recompile with CUDA and CNN etc but I've had problems with them (problems with stbConfig.cmake).
This is the output of the training. Can anybody help me please?
Thanks, Silvia
![IMG_20190819_153522](https://user-images.githubusercontent.com/45975704/63270514-97da6280-c298-11e9-9710-794310f3d354.png)

**Comment by lsd1994** [comment]
Add -dont_show have a try.

**Comment by silviasanna** [comment]
@Isd1994 it made something but after 3 lines it killed again. How can I do? Could it be that I haven't enough ram?

**Comment by silviasanna** [comment]
@Isd1994 it made something but after 3 lines it killed again. How can I do? Could it be that I haven't enough ram?

**Comment by lsd1994** [comment]
Set gpu=0 and make again, if ok, maybe something wrong on your gpu configuration, if not, maybe something wrong on your darknet.

**Comment by silviasanna** [comment]
Thank you, I've solved with the swap (I made a bigger file because I had a lot of images 10GB and my RAM was only 8)
Now I'm training but I have IOU Nan, GIOU Nan Class Nan Obj Nan. Is it normal or is something wrong that I can fix? 
Thank you all

**Comment by lsd1994** [comment]
After training a batch, avg loss shouldn't be nan, but nan on other position (include iou,class...) is normal.

**Comment by silviasanna** [comment]
Thank you a lot! I have one more question: with validation images, does it mean the test images?

**Comment by lsd1994** [comment]
Yes. yolo use valid=xxx.txt in .data file to calculate mAP while training.

**Comment by silviasanna** [comment]
Ok, I have launched the command for the map but my output is: "Can't open label file. (This can be normal only if you use MSCOCO): [path to the image test]". Is it normal or I made some mistakes? 
For example the guide says to have file yolo-obj_7000.weights yolo-obj_8000.weights yolo-obj_9000.weights but I only have one with 8678559 weights (I've stopped the train at 256 iteration because the average loss was good enough).
In the calculation of mAP shall I see all the images test with the rectangular?

**Comment by silviasanna** [comment]
Hi anyone, I have troubles with the yolov3-tiny_obj_last.weights file. I've created the folder backup and put the correct path in the obj.data file in the directory data, but in the directory backup I haven't the file. Can anyone help me please? Thanks

**Comment by lsd1994** [comment]
> Ok, I have launched the command for the map but my output is: "Can't open label file. (This can be normal only if you use MSCOCO): [path to the image test]". Is it normal or I made some mistakes?
jpg file and txt should in the same folder.



> Hi everyone, I have troubles with the yolov3-tiny_obj_last.weights file. I've created the folder backup and put the correct path in the obj.data file in the directory data, but in the directory backup I haven't the file. Can anyone help me please? Thanks
Please show the output of darknet.

**Comment by silviasanna** [comment]
"jpg file and txt should in the same folder."
Sorry I was wrong, the problem is in the opening of the labels folder. In that label I only have images of letters and punctuation. I suppose that I should be have images about my labels. Is it right? In that case how can I do?


For the problem of the weights file I've solved, it was automatically created after 100 iteractions.
